cdk/src/open-data-platform/data-plane/api/api-stack.ts

```diff
@@ -0,0 +1,64 @@
+import { Duration } from 'aws-cdk-lib';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as lambda from 'aws-cdk-lib/aws-lambda-nodejs';
+import * as apigateway from 'aws-cdk-lib/aws-apigateway';
+import { Construct } from 'constructs';
+import { SchemaProps } from '../schema/schema-props';
+import * as path from 'path';
+
+export class ApiStack extends Construct {
+  constructor(scope: Construct, id: string, props: SchemaProps) {
+    super(scope, id);
+
+    const { cluster, vpc, db, credentialsSecret } = props;
+
+    const lambdaFunction = new lambda.NodejsFunction(this, 'geolocate-handler', {
+      entry: `${path.resolve(__dirname + '/../../../../../')}/api/src/geolocate/get.handler.ts`,
+      handler: 'handler',
+      vpc: vpc,
+      vpcSubnets: { subnetType: ec2.SubnetType.PRIVATE_WITH_NAT },
+      environment: {
+        CREDENTIALS_SECRET: credentialsSecret.secretArn,
+        DATABASE_NAME: db,
+        RESOURCE_ARN: props.cluster.clusterArn,
+      },
+      memorySize: 512,
+      timeout: Duration.minutes(15),
+      bundling: {
+        externalModules: ['aws-sdk'],
+        nodeModules: [],
+      },
+    });
+
+    credentialsSecret.grantRead(lambdaFunction);
+    cluster.connections.allowFrom(lambdaFunction, ec2.Port.tcp(cluster.clusterEndpoint.port));
+
+    if (lambdaFunction.role?.roleArn != undefined) {
+      let role = iam.Role.fromRoleArn(
+        scope,
+        id + '-' + lambdaFunction,
+        lambdaFunction.role.roleArn,
+      );
+      cluster.grantDataApiAccess(role);
+    }
+
+    const api = new apigateway.RestApi(this, 'open-data-platform-api', {
```

Comment:
[Can you use this](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.FunctionUrlOptions.html) to create a URL without setting up API Gateway?

Reply:
Decided offline: going to check this in to unblock FE work. Brett's going to look into converting from API gateway --> cloudfront to get a domain for this api instead of creating a new domain